src/components/Post.jsx

```diff
@@ -202,81 +205,85 @@ function Post(prop) {
       className="post"
       style={{ boxShadow: "0px 0px 5px 1px rgba(0, 0, 0, 0.4)" }}
     >
-      <div className="post__header">
-        <Avatar
-          className="post__avatar"
-          alt={username}
-          src={avatar}
-          sx={{
-            bgcolor: "royalblue",
-            border: "2px solid transparent",
-            display: "flex",
-            justifyContent: "center",
-            alignItems: "center",
-            cursor: "pointer",
-            "&:hover": {
-              boxShadow: "rgba(100, 100, 111, 0.2) 0px 7px 17px 0px",
-              border: "2px solid black",
-              scale: "1.1",
-            },
-          }}
-        />
-        <h3 className="post__username">{username}</h3>
-        <div className="social__icon__last">
-          <IconButton
-            aria-label="more"
-            id="long-button"
-            aria-controls={open ? "long-menu" : undefined}
-            aria-expanded={open ? "true" : undefined}
-            aria-haspopup="true"
-            onClick={(event) => setAnchorEl(event.currentTarget)}
+      <Link
```

Comment:
why are you wrapping the whole post head as link, this would restrict us from interacting with the post properly(clicking on user profile, trying to delete or other options wont work)